From 8b74fa8bd827126a363cd5c780b6940e438cdd02 Mon Sep 17 00:00:00 2001
From: Desnes Nunes <desnesn@linux.ibm.com>
Date: Wed, 18 May 2022 14:12:38 -0300
Subject: [PATCH 21/23] beaker: dn: ubuntu: fix harness address and download CA crt

This makes harness address generic and fetches the CA crt during installation

Signed-off-by: Paul A. Clarke <pc@us.ibm.com>
Signed-off-by: Desnes Nunes <desnesn@linux.ibm.com>
---
 Server/bkr/server/autoinstalls/default | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Server/bkr/server/autoinstalls/default b/Server/bkr/server/autoinstalls/default
index 1103ff641..84fa5b165 100644
--- a/Server/bkr/server/autoinstalls/default
+++ b/Server/bkr/server/autoinstalls/default
@@ -34,7 +34,9 @@ autoinstall:
     - echo -n >/target/etc/profile.d/task-overrides-rhts.sh
     - echo "export RHTS_OPTION_COMPATIBLE=">>/target/etc/profile.d/task-overrides-rhts.sh
     - echo "export RHTS_OPTION_COMPAT_SERVICE=">>/target/etc/profile.d/task-overrides-rhts.sh
-    - echo "deb [allow-insecure=yes trusted=yes] http://FQDN/harness/{{ distro.osversion.osmajor }} /" >> /target/etc/apt/sources.list
+    - echo "deb [trusted=yes] {{ absolute_url('/') }}harness/{{ distro.osversion.osmajor }} /" >> /target/etc/apt/sources.list
+    - wget --no-check-certificate -r -l1 -np -nH -e robots=off -A '*crt' --cut-dirs=2 -P /target/usr/local/share/ca-certificates/ {{ absolute_url('/certs') }}
+    - curtin in-target --target=/target -- update-ca-certificates
     - curtin in-target --target=/target -- apt update
     - curtin in-target --target=/target -- apt install make restraint restraint-rhts
     - curtin in-target --target=/target -- systemctl enable restraintd
-- 
2.35.3

